iohub/dataset_base.py

```diff
@@ -107,37 +108,69 @@ def __eq__(self, other: BaseFOV) -> bool:
         return self._root.absolute() == other._root.absolute()
 
 
-class BaseFOVCollection(Mapping):
-    @abstractmethod
-    def __enter__(self) -> BaseFOVCollection:
-        """Open the underlying file and return self."""
-        raise NotImplementedError
+class FOVCollection(Mapping):
+    """
+    Basic implementation of a mapping of strings to BaseFOVs.
+    """
 
-    @abstractmethod
-    def __exit__(
+    def __init__(
         self,
-        exc_type: Optional[Type[BaseException]],
-        exc_val: Optional[BaseException],
-        exc_tb: Optional[TracebackType],
-    ) -> bool:
-        """Close the files."""
-        raise NotImplementedError
+        data_dict: Optional[dict[str, BaseFOV]] = None,
```

Comment:
This constructor needs reference to all of the FOVs at creation time, meaning that for HCS OME-Zarr the entire tree has to be walked upon reading the plate. Is there a way to do this lazily as it is currently?

Reply:
HCS OME-Zarr could have its own constructor and provide no parameter do the parent `FOVCollection` class, or do you prefer an abstract base class that doesn't initialize anything?

Reply:
But then the attributes initialized are expected elsewhere, which makes it difficult to maintain. I think the base class should be abstract and the implementation here can become something like `FOVDict`